pkg/api/policy/checks.go

```diff
@@ -0,0 +1,348 @@
+// Copyright (c) 2018 Huawei Technologies Co., Ltd. All Rights Reserved.
+//
+//    Licensed under the Apache License, Version 2.0 (the "License"); you may
+//    not use this file except in compliance with the License. You may obtain
+//    a copy of the License at
+//
+//         http://www.apache.org/licenses/LICENSE-2.0
+//
+//    Unless required by applicable law or agreed to in writing, software
+//    distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+//    WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+//    License for the specific language governing permissions and limitations
+//    under the License.
+
+//
+
+package policy
+
+import (
+	"fmt"
+	"reflect"
+	"regexp"
+	"strings"
+
+	log "github.com/golang/glog"
+	"github.com/opensds/opensds/pkg/utils"
+)
+
+type NewCheckFunc func(kind string, match string) BaseCheck
+
+var registeredChecks map[string]NewCheckFunc
+
+func register(name string, fun NewCheckFunc) {
```

Comment:
It's weired to create a variable called `fun`, it's better if you change it to `f` or `cf`.